### PR TITLE
search, fromstix test, elastic function

### PR DESCRIPTION
**Pull request by noaakl**

### CODE REVIEW

**Comment by barvhaim** [comment]
@noaakl Reproduce bug in To STIX -
1. create two objects `A` and `B`
2. make reference from object `A` field to `B`
3. delete object `B`
4. reference is stuck, and if you save the map it is appear in there

File: src/components/ToSTIX/to_stix.module.scss [hunk 1]
```diff
@@ -46,6 +51,20 @@
...
+.modal {
+  padding: {
+    top: $spacing-03;
+    bottom: $spacing-10;
+  }
+  margin-top: $spacing-03;
+  margin-bottom: $spacing-13;
+}
+
+.ComboBox {
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: please use the same code styling as other fields `.combo_box`

File: src/components/ToSTIX/to_stix.module.scss [hunk 2]
```diff
@@ -46,6 +51,20 @@
...
+  background-color: white;
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: `$carbon--white-0`

File: src/components/ToSTIX/to_stix.module.scss [hunk 3]
```diff
@@ -64,3 +83,7 @@
     @include carbon--type-style("label-01");
   }
 }
+
+.background {
+  background-color: white;
```
Comment (barvhaim) [inline src/components/ToSTIX/to_stix.module.scss]: `$carbon--white-0`

File: src/components/ToSTIX/StixField.jsx [hunk 4]
```diff
@@ -56,13 +58,13 @@ const StixField = ({
...
         </div>
         <div className={"bx--col-md-2"}>
-          <Dropdown
+          <ComboBox
+            id={`ComboBox_${stixFieldId}`}
             size={"sm"}
-            ariaLabel="Dropdown"
-            id="transformer"
+            placeholder={"Search Transformer"}
+            ariaLabel="ComboBox"
```
Comment (barvhaim) [inline src/components/ToSTIX/StixField.jsx]: transformers_combobox

File: src/components/ToSTIX/SourceField.jsx [hunk 5]
```diff
@@ -30,10 +37,28 @@ const SourceFieldHeader = ({ fieldId, objectKey, fieldData }) => {
           onChange={(e) => {
             dispatch(updateDataSourceField(objectKey, fieldId, e.target.value));
           }}
-          value={fieldData.field}
+          value={fieldName}
           size={"sm"}
         />
       </div>
+      <div>
+        <WatsonHealthStackedMove20
```
Comment (barvhaim) [inline src/components/ToSTIX/SourceField.jsx:45]: I think it's called just `StackedMove20` 
image

Comment (noaakl) [reply src/components/ToSTIX/SourceField.jsx:45]: I thought so too, but to use it we have to use WatsonHealthStackedMove20

File: src/components/ToSTIX/StixFieldsTable.jsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ const StixFieldsTableBody = ({ objectKey, sourceFieldId, stixFieldsData }) => {
     const stixFieldTransformer = stixField?.transformer;
     const stixFieldReferences =
       stixField.references && stixField.references.length !== 0
-        ? !typeof stixField.references === Array
+        ? stixField.references.constructor === Array
           ? stixField.references
           : [stixField.references]
```
Comment (barvhaim) [inline src/components/ToSTIX/StixFieldsTable.jsx]: when is this case happening?

**Review by barvhaim — APPROVED** [review]

**Pull request merged**